dtp/src/utils.rs

```diff
@@ -200,6 +202,52 @@ pub(crate) async fn multicall_light_txs_1(
     Ok(())
 }
 
+/// Handle future calls by batching method into a batch of max. chunk size.
+/// Otherwise, without batching, it's failing when requested too many connections at once.
+/// All new accounts are incrementing numbers (as considered this activity).
+/// NOTE: There are many combination of running these txs by multiple signers.
+/// But, for simplicity here, we have considered only `increment` function call
+/// of `Counter` contract.
+///
+/// Here, instead of sending `calls` at once via `join_all(calls).await` as shown here:
+/// ```rust
+/// let mut calls = Vec::new();
+/// for signer in &signers {
+///     // collect all contract setter calls into a vec of futures. So, not awaited on each future in this loop.
+///     calls.push(counter_increment(client.to_owned(), counter_address, signer.to_owned()));
+/// }
+/// // async calls awaited all at once so as to try to put as many txs into a/few blocks than
+/// // sending each tx into each block.
+/// // Here, senders are different for each tx. So, not a problem of dependency on each other.
+/// join_all(calls).await;
+/// ```
+///
+/// Use `batch` to run each batch via `join_all(batch).await`. E.g. for 1000 connections,
+/// there would be 10 batches of 100 calls/requests each. Now, each batch i.e. 100 requests
+/// is sent at once, unlike all 1000 (total) calls sent at once as was done previously.
+async fn handle_async_calls_in_batch(
+    client: Arc<Provider<Http>>,
+    counter_address: Address,
+    signers: Vec<Wallet<SigningKey>>,
+) -> eyre::Result<()> {
+    // TODO: add chunk_size related code
+
+    // iteration in chunk of `MAX_BATCH_CHUNK_SIZE`
+    for chunk in signers.chunks(MAX_BATCH_CHUNK_SIZE.into()) {
+        // create a batch vec for this chunk
+        let mut batch = Vec::new();
+
+        for signer in chunk {
+            batch.push(counter_increment(client.to_owned(), counter_address, signer.to_owned()));
```

Comment:
You could generalize this to take in a function to batch. Can be done later.

Reply:
Yeah, refactoring is pending once all the features are added.